### ISSUE TITLE
Configurability for line endings in textgen

### ISSUE THREAD

**Issue by eugenschindler** [JetBrains/MPS-extensions#186]
From mps-plaintextgen created by [eugenschindler](https://github.com/eugenschindler): DSLFoundry/mps-plaintextgen#40  

Make CR, LF and CRLF configuration options for all line endings. 
Consider having a configuration for an entire model, solution or project and not only for one template.
The '\n'  that needs to be made configurable to "\r" (CR), "\n" (LF) or "\r\n" (CRLF) is located in CharacterMatrix.ToString(row).

**Comment by eugenschindler** [comment]
To have maximum configurability and flexibility, I think it would be good to have plaintextgen give an option to the language developer to determine where the CR/LF settings can be customized. The developer should specify one or more of the following points for customizing the setting:
 1. Hard-coded on template-level (enforcing only the setting determined by the language developer).
 2. Application- or project-specific (allowing users to override default-settings for specific languages specified by language developer).
 3. Model- or root node-specific (allowing users to override settings per relevant input unit, similar to mbeddr build configurations).

In addition, it should be possible to specify which settings take precedence in which order in case several options are allowed and indeed set.

@klemensschindler and @JoostvanPinxten, what do you think about this? Of course, others are also very welcome to chip in.

**Comment by jennek** [comment]
My view: approach 1 is good enough for now. It will solve the issue I now have. I have no need to switch end-of-line within a single text file -- I doubt anyone ever will. So a setting per generated text file is right for me. Default settings on the application or project or user level seem nice to have but for me it is too early to tell what the real need (use case) is.

**Comment by JoostvanPinxten** [comment]
I agree with Jennek. At the moment I feel it is sufficient for most to have a template-level configurability. Although maximum flexibility and support would be nice, in this case I would just enable the most urgent cases; i.e. per-file generated template-level configurability.

Perhaps an interesting next step is to allow the template-level line-ending style to be a derived/queried property from the input model? (Haven't had a look at the actual implementation you have recently merged.)
